test_tools/healthcheck_test_tool/main.go

```diff
@@ -151,24 +158,46 @@ func main() {
 		log.Fatalf("Invalid IP address: %v", *ip)
 	}
 
-	switch *hcType {
-	case "dns":
-		doDNSCheck(target)
-	case "http":
-		doHTTPCheck(target, false)
-	case "https":
-		doHTTPCheck(target, true)
-	case "ping":
-		doPingCheck(target)
-	case "radius":
-		doRADIUSCheck(target)
-	case "tcp":
-		doTCPCheck(target, false)
-	case "tcp_tls":
-		doTCPCheck(target, true)
-	case "udp":
-		doUDPCheck(target)
-	default:
-		log.Fatalf("Unsupported healthcheck type: %q", *hcType)
+	errs := make(chan error, *parallel)
+
+	for i := 0; i < *parallel; i++ {
+		go func(err chan error) {
+			switch *hcType {
+			case "dns":
+				err <- doDNSCheck(target)
+			case "http":
+				err <- doHTTPCheck(target, false)
+			case "https":
+				err <- doHTTPCheck(target, true)
+			case "ping":
+				err <- doPingCheck(target)
+			case "radius":
+				err <- doRADIUSCheck(target)
+			case "tcp":
+				err <- doTCPCheck(target, false)
+			case "tcp_tls":
+				err <- doTCPCheck(target, true)
+			case "udp":
+				err <- doUDPCheck(target)
+			default:
+				log.Fatalf("Unsupported healthcheck type: %q", *hcType)
+			}
+		}(errs)
 	}
+	count := 0
+	fail := 0
+	for {
```

Comment:
how do you exit from the loop?

Reply:
Sorry, I miss some lines when I submitted the PR. I will add.